Frameworks/CoreData/NSManagedObjectModel-XMLParsing.mm

```diff
@@ -0,0 +1,282 @@
+//******************************************************************************
+//
+// Copyright (c) 2016 Microsoft Corporation. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#import <CoreData/CoreData.h>
+#import <CoreFoundation/CFXMLParser.h>
+#import <CoreFoundation/CoreFoundation.h>
+#import <Foundation/Foundation.h>
+
+#import <ErrorHandling.h>
+#import <LoggingNative.h>
+#import <Starboard/SmartTypes.h>
+
+#import "NSManagedObjectModel-XMLParsing.h"
+
+#include <list>
+#include <tuple>
+
+static NSString* const _kNSManagedObjectModelTag = @"model";
+static NSString* const _kNSManagedObjectModelEntityTag = @"entity";
+static NSString* const _kNSManagedObjectModelEntityAttributeTag = @"attribute";
+static NSString* const _kNSManagedObjectModelEntityRelationshipTag = @"relationship";
+static NSString* const _kNSManagedObjectModelUserInfoTag = @"userInfo";
+static NSString* const _kNSManagedObjectModelUserInfoEntryTag = @"entry";
+static NSString* const _kNSManagedObjectModelEntityCompoundIndexTag = @"compoundIndex";
+static NSString* const _kNSManagedObjectModelEntityCompoundIndexesTag = @"compoundIndexes";
+static NSString* const _kNSManagedObjectModelEntityCompoundIndexIndexTag = @"index";
+static NSString* const _kNSManagedObjectModelEntityUniquenessConstraintTag = @"uniquenessConstraint";
+static NSString* const _kNSManagedObjectModelEntityUniquenessConstraintsTag = @"uniquenessConstraints";
+static NSString* const _kNSManagedObjectModelEntityUniquenessConstraintConstraintTag = @"constraint";
+
+// To save ourselves some string compares:
+static NSString* _uniqueTag(NSString* tag) {
+    static StrongId<NSSet> _tagSet = [NSSet setWithObjects:_kNSManagedObjectModelTag,
```

Comment:
I understand now: GitHub's e-mail did not include the `<NSString*>` (because HTML :P) and I thought you were asking why it was an NSSet.
This is a great suggestion. Thanks. :)

---
In reply to: [70035761](https://github.com/Microsoft/WinObjC/pull/610#discussion_r70035761) [](ancestors = 70035761)